mbed_flasher/flashers/FlasherST.py

```diff
@@ -94,6 +97,17 @@ def flash(self, source, target, method, no_reset, target_filename=None):
         :param target_filename: target filename (not in use)
         :return: 0 when flashing success
         """
+        tmp_file = None
+        try:
+            if target_filename:
+                tmp_file = tempfile.NamedTemporaryFile(delete=False, suffix=target_filename)
+                with open(source, 'rb') as source_file:
+                    tmp_file.write(source_file.read())
+                source = tmp_file.name

```

Comment:
Do you need to close the file to make sure all buffers are flushed and all writes are actually done to the disk?

Reply:
Yes.